vaadin-grid-flow-parent/vaadin-grid-flow/src/main/resources/META-INF/resources/frontend/gridConnector.js

```diff
@@ -786,44 +780,40 @@ import { GridFlowSelectionColumn } from "./vaadin-grid-flow-selection-column.js"
             cache[parentKey][0] = [];
           }
 
-          // If grid has outstanding requests for this parent, then resolve them
-          // and let grid update the flat size and re-render.
-          let outstandingRequests = Object.getOwnPropertyNames(treePageCallbacks[parentKey] || {});
-          for (let i = 0; i < outstandingRequests.length; i++) {
-            let page = outstandingRequests[i];
-
-            let lastRequestedRange = lastRequestedRanges[parentKey] || [0, 0];
-
-            const callback = treePageCallbacks[parentKey][page];
-            if (
-              (cache[parentKey] && cache[parentKey][page]) ||
-              page < lastRequestedRange[0] ||
-              page > lastRequestedRange[1]
-            ) {
-              delete treePageCallbacks[parentKey][page];
-              let items = cache[parentKey][page] || new Array(levelSize);
-              callback(items, levelSize);
-            } else if (callback && levelSize === 0) {
-              // The parent item has 0 child items => resolve the callback with an empty array
-              delete treePageCallbacks[parentKey][page];
-              callback([], levelSize);
-            }
-          }
+          const parentItem = createEmptyItemFromKey(parentKey);
+          const parentItemContext = dataProviderController.getItemContext(parentItem);
+          const parentItemSubCache = parentItemContext?.subCache;
+          if (parentItemSubCache) {
+            // If grid has outstanding requests for this parent, then resolve them
```

Comment:
nit: we could use the term "pending" instead of "outstanding" in the connector (3 places) especially now since the variables named `outstandingRequests` have been removed.

Reply:
Good point, done.